src/activities/jumbled_words/jumbled_words.js

```diff
@@ -0,0 +1,177 @@
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+.pragma library
+.import QtQuick 2.0 as Quick
+.import GCompris 1.0 as GCompris
+.import QtGraphicalEffects 1.0 as QtGraphicalEffects
+
+.import "qrc:/gcompris/src/core/core.js" as Core
+
+var url = "qrc:/gcompris/src/activities/jumbled_words/resource/"
+var url1 = "qrc:/gcompris/src/activities/lang/resource/words_sample.json"
+var defaultLevelsFile = "qrc:/gcompris/src/activities/lang/resource/words_sample.json";
+
+var currentLevel
+var maxLevel
+var maxSubLevel
+var currentSubLevel
+var items
+var levels;
+var level;
+var temp;
+var copy;
+var questions;
+var answers;
+var items;
+var generate;
+var array = [];
+
+function start(items_) {
+
+    items = items_
+    loadLevels();
+    currentLevel = 0;
+    currentSubLevel = 0;
+    maxLevel = levels.length;
+    initLevel()
+}
+
+function loadLevels()
+{
+    var filename = GCompris.ApplicationInfo.getLocaleFilePath(url1);
+    levels = items.parser.parseFromUrl(filename);
+    if (levels == null) {
+        console.warn("Jumbled_words: Invalid levels file " + filename);
+        levels = items.parser.parseFromUrl(defaultLevelsFile);
+        if (levels == null) {
+            console.error("Jumbled_words: Invalid default levels file "
+                + defaultLevelsFile + ". Can't continue!");
+            return;
+        }
+    }
+}
+
+function stop() {
+
+}
+
+function shuffleString(s)
+{
+    var a = s.split("");
+    var n = a.length;
+
+    for(var i = n-1; i>0; i--) {
+        var j = Math.floor(Math.random() * (i + 1));
+        var tmp = a[i];
+        a[i] = a[j];
+        a[j] = tmp;
+    }
+    return a.join("");
+}
+
+function initLevel() {
+    items.bar.level = currentLevel + 1
+    if (currentSubLevel == 0) {
+        level = levels[currentLevel];
+        maxSubLevel = level.content[0].content.length;
+        for(var i=0;i< maxSubLevel ; i++)
+        {
+            array[i]=i;
+        }
+        array=Core.shuffle(array);
+        items.score.numberOfSubLevels = maxSubLevel;
+        items.score.currentSubLevel = "1";
+        temp=array[currentSubLevel];
+        answers = level.content[0].content[temp].description;
+        copy = answers;
+        var demo = shuffleString(answers);
+        if(demo == answers)
+        {
+            demo = shuffleString(answers);
+        }
+        else
+            questions = demo;
+    }
+    else {
+        do{
+            temp=temp=array[currentSubLevel];
+            generate = level.content[0].content[temp].description;
+
+        } while(copy===generate);
+        copy=generate;
```

Comment:
what is the use of copy, generate...? You seem to have a lot of variables doing more or less the same.